api/server/httpstatus/status.go

```diff
@@ -150,3 +155,14 @@ func statusCodeFromContainerdError(err error) int {
 		return http.StatusInternalServerError
 	}
 }
+
+func statusCodeFromLibnetworkError(err error) int {
+	switch {
+	case nettypes.IsBadRequestError(err):
```

Comment:
can we change the libnetwork errors to match the `errdefs` signature?

Reply:
There're only 10 structs implementing `BadRequestError` and `NoServiceError` so it should be easy to do. However, could you share why you prefer that over the current implementation?

Reply:
The idea behind the errdefs package was to define a set of interfaces/types that can be implemented by packages in our code. Using those types means that code "somewhere in the stack" can produce an error of a specific "errdefs" type, and any other code consuming it is able to use that information (e.g. to check if the error was an `errdefs.IsNotFound()`); any other error (baring some errors we'd get from Go's stdlib, or when _directly_ dealing with sentinel errors) can be considered "just an error".

I'd rather not make this package handle specific implementations of "other" type of errors beyond the types that are defined by `errdefs`. Ultimately, we should also get rid of the special cases for containerd / grpc errors and the distribution errors; we should make sure that such errors are converted to a proper `errdefs` type (where needed) before they propagate through our code, but I think at the time there were still too many paths to check in that respect, so we left that for a "later exercise"

Now that libnetwork is in this repository (no longer an external repository), and we have a better grip on what kind of errors can be returned (and when), we should update the libnetwork types to implement `errdefs` types (where needed - i.e., where their error should be considered a specific type).